contracts/libraries/OracleLibrary.sol

```diff
@@ -76,4 +76,35 @@ library OracleLibrary {
 
         return uint32(block.timestamp) - observationTimestamp;
     }
+
+    /// @notice Given a pool, it returns the tick value as of the start of the current block
+    /// @param pool Address of Uniswap V3 pool
+    /// @return The tick that the pool was in at the start of the current block
+    function getBlockStartingTick(address pool) internal view returns (int24) {
+        (, int24 tick, uint16 observationIndex, uint16 observationCardinality, , , ) = IUniswapV3Pool(pool).slot0();
+        require(observationCardinality > 0, 'PNI');
+
+        // If the latest observation occurred in the past, then no tick-changing trades have happened in this block
+        // therefore the tick in `slot0` is the same as at the beginning of the current block.
+        // We don't need to check if this observation is initialized - it is guaranteed to be.
+        (uint32 observationTimestamp, , , ) = IUniswapV3Pool(pool).observations(observationIndex);
+        if (observationTimestamp < block.timestamp) {
+            return tick;
+        }
+
+        // We need to fetch the two prior observations and use them to calculate the tick.
+        require(observationCardinality > 2, 'NEO');
```

Comment:
you can return the tick written to the latest observation, which was the tick at the start of the block